File: java-vfs/src/main/java/emg/java/vfs/commands/Mkdir.java
package emg.java.vfs.commands;

import emg.java.vfs.files.DirEntry;
import emg.java.vfs.files.Directory;
import emg.java.vfs.filesystem.State;

public class Mkdir extends CreateEntry {

    public Mkdir(String name) {
        super(name);
    }

    @Override
    public DirEntry createSpecificEntry(State state, String entryName) {
        return Directory.empty(state.wd().path(), entryName);
    }
}

File: kotlin-vfs/src/main/kotlin/emg/kotlin/vfs/commands/Mkdir.kt
package emg.kotlin.vfs.commands

import emg.kotlin.vfs.files.DirEntry
import emg.kotlin.vfs.files.Directory
import emg.kotlin.vfs.filesystem.State

class Mkdir(name: String) : CreateEntry(name) {
    override fun createSpecificEntry(state: State, entryName: String): DirEntry {
        return Directory.empty(state.wd.path(), entryName)
    }
}